clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -70,6 +75,17 @@ public AbstractConfig(ConfigDef definition, Map<?, ?> originals) {
         this(definition, originals, true);
     }
 
+    /**
+     * Called directly after user configs got parsed (and thus default values got set).
+     * This allows to change default values for "secondary defaults" if required.
+     *
+     * @param parsedValues unmodifiable map of current configuration
+     * @return a map of updates that should be applied to the configuration (will be validated to prevent bad updates)
+     */
+    protected Map<String, Object> postProcessParsedConfig(Map<String, Object> parsedValues) {
```

Comment:
What I was thinking is actually different from the original solution: in `getCommonConsumerProducerConfigs` we check on the EOS config value, and then depending on that modify the streams overridden defaults before user's overrides:

```
        final Map<String, Object> consumerProps = new HashMap<>(CONSUMER_DEFAULT_OVERRIDES);

        consumerProps.putAll(EOS_CONSUMER_DEFAULT_OVERRIDES); // this is added line.

        consumerProps.putAll(clientProvidedProps);
```

Does it make sense?